File: apps/inventory/frontend/src/services/product.js
import http from './http'

export function getAll() {

  return http.get('/api/products/')
}

export function getById(id) {

  return http.get(`/api/products/${id}/`)
}

export function create(data) {

  return http.post('/api/products/', data)
}

export function update(id, data) {

  return http.put(
    `/api/products/${id}/`,
    data
  )
}

export function remove(id) {

  return http.delete(
    `/api/products/${id}/`
  )
}

File: apps/inventory/services/product.py
from ..models.product import Product

class ProductService:
    """
    Business logic layer for Product.
    Decouples database operations from views.
    """
    @staticmethod
    def get_all():
        return Product.objects.all()

    @staticmethod
    def get_by_id(obj_id):
        return Product.objects.filter(id=obj_id).first()

    @staticmethod
    def create(data):
        return Product.objects.create(**data)

    @staticmethod
    def update(obj_id, data):
        obj = Product.objects.filter(id=obj_id).first()
        if obj:
            for key, value in data.items():
                setattr(obj, key, value)
            obj.save()
        return obj

    @staticmethod
    def delete(obj_id):
        obj = Product.objects.filter(id=obj_id).first()
        if obj:
            obj.delete()
            return True
        return False
